### ISSUE TITLE
[Metric] use Float64Counter to get wrong metric data

### ISSUE THREAD

**Issue by gunsluo** [open-telemetry/opentelemetry-go#914]
**Describe the bug**
use Float64Counter to get the wrong metric data.

the code
```
meterPusher = push.New(
			simple.NewWithExactDistribution(),
			exporter,
			push.WithPeriod(30*time.Second),
			//push.WithTimeout(10*time.Second),
		)
meterProvider = meterPusher.Provider()
meter:= meterProvider.Meter("name")

accountReadCounter : =meter.NewFloat64Counter("account.read", metric.WithDescription("record number of reading account"));

....

accountReadCounter.Add(ctx, 1)

```

the counter metric's value should not be reduced.

![image](https://user-images.githubusercontent.com/9067076/86736023-c9514d80-c065-11ea-963b-4e80950f0e53.png)

**What did you expect to see?**
the value of the counter keeps increasing by time.

**What version did you use?**
Version:  
collector 0.5.0 
otel: 0.7.0

**What config did you use?**
collector Config: 
```
receivers:
  otlp:
    protocols:
      grpc:
        endpoint: "0.0.0.0:55679"

processors:
  batch:
  queued_retry:

extensions:
  health_check: {}

exporters:
  jaeger:
    endpoint: "jaeger:14250"
    insecure: true
  prometheus:
    endpoint: "0.0.0.0:8889"
    namespace: example
    const_labels:
      label1: test
  logging:
      loglevel: debug

service:
  extensions: [health_check]
  pipelines:
    traces:
      receivers: [otlp]
      processors: [batch, queued_retry]
      exporters: [jaeger]

    metrics:
      receivers: [otlp]
      exporters: [prometheus, logging]
```

**Environment**
OS:  macOS 
Compiler: go version go1.14.2 darwin/amd64

**Additional context**
it works fine on version 0.3.0(collector) and otel(0.6.0).

**Comment by gunsluo** [comment]
Hi, @nilebox 

I found that SDK reset the value of the counter when record's updateCount is not collectedCount,   there is no any modification of collectedCount in code.   so the value of the counter is reset.

https://github.com/open-telemetry/opentelemetry-go/blob/master/sdk/metric/sdk.go#L376 

https://github.com/open-telemetry/opentelemetry-go/blob/27e892aad23241c19a55155fedcc980bc9db551e/sdk/metric/sdk.go#L437


is it a bug?

**Comment by Aneurysm9** [comment]
@jmacd is this related to #887/#903?

**Comment by jmacd** [comment]
I am a bit confused. From the report, it looks more like version 0.6 is still being used. The 0.7 release added metrics processor logic to sum the individual counts, and the problems in #887 and #903 would have produced different odd behavior. Is there any chance this is a 0.6 otel library with a 0.4 collector? Otherwise, it's an unusual report and I would like to continue investigating.

**Comment by jmacd** [comment]
This is a case of https://github.com/open-telemetry/opentelemetry-collector/issues/1255. The OTLP to Prometheus functionality is broken in the 0.4 collector release: we will disable the OTLP receiver in the 0.5 release and get this fixed in 0.6 I hope.

**Comment by jmacd** [comment]
Closing this as it's covered in collector issue 1255.
